fdbserver/networktest.actor.cpp

```diff
@@ -24,6 +24,44 @@
 
 UID WLTOKEN_NETWORKTEST( -1, 2 );
 
+struct latency_stats {
+	using sample = double;
+	double x = 0;
+	double x2 = 0;
+	double n = 0;
+
+	sample tick() {
+		// now() returns the timestamp when we were scheduled; count
+		// all that time against this sample.
+		return now();
+	}
+
+	void tock(sample tick) {
+		// time_monotonic returns the timestamp when it was called;
+		// count the time it took us to be dispatched and invoke
+		// timer_monotonic
+		double delta = timer_monotonic() - tick;
+		x += delta;
+		x2 += (delta * delta);
+		n++;
+	}
+
+	void reset() {
+		*this = latency_stats();
+	}
+
+	double count() {
+		return n;
+	}
+
+	double mean() {
+		return x/n;
+	}
+	double stddev() {
+		return sqrt(x2/n - (x/n)*(x/n));
```

Comment:
Are you doing (x/n)*(x/n) instead of x2/n/n because of floating point rounding?

Reply:
I wrote it this way for clarity (the formula is (x/n)^2).  I suspect it generates slightly faster code (R1=x/n; R2=R1*R1 is only two floating points ops instead of three), but that doesn't really matter here.

Also, note that x2 != x^2, so I'd have to write x*x/n/n.